airbyte-scheduler/scheduler-persistence/src/main/java/io/airbyte/scheduler/persistence/DefaultJobCreator.java

```diff
@@ -102,8 +95,17 @@ public Optional<Long> createResetConnectionJob(final DestinationConnection desti
       throws IOException {
     final ConfiguredAirbyteCatalog configuredAirbyteCatalog = standardSync.getCatalog();
     configuredAirbyteCatalog.getStreams().forEach(configuredAirbyteStream -> {
-      configuredAirbyteStream.setSyncMode(SyncMode.FULL_REFRESH);
-      configuredAirbyteStream.setDestinationSyncMode(DestinationSyncMode.OVERWRITE);
+      final StreamDescriptor streamDescriptor = new StreamDescriptor()
+          .withName(configuredAirbyteStream.getStream().getName())
+          .withNamespace(configuredAirbyteStream.getStream().getNamespace());
+      if (streamsToReset.contains(streamDescriptor)) {
+        configuredAirbyteStream.setSyncMode(SyncMode.FULL_REFRESH);
+        configuredAirbyteStream.setDestinationSyncMode(DestinationSyncMode.OVERWRITE);
+      } else {
+        // set sync mode to full_refresh/append so that these streams are NOT affected in the destination
+        configuredAirbyteStream.setSyncMode(SyncMode.FULL_REFRESH);
```

Comment:
This logic means that any streams passed into the `streamsToReset` argument will be set to full_refresh/overwrite, and any other streams in the catalog will be full_refresh/append.

The plan is for the API to pass any new or updated streams into `streamsToReset`, so those will be set to overwrite. Any other streams in the catalog (I think this will just be the non-updated non-new streams, because deleted ones won't be in the catalog) will be append and the empty airbyte source shouldn't emit any records for those, so I believe that should result in no change in the destination for those streams.